packages/serverpod/lib/src/server/websocket_request_handlers/helpers/method_stream_manager.dart

```diff
@@ -571,21 +578,13 @@ class MethodStreamManager {
     }
 
     outputController.addStream(methodStream).whenComplete(
-      () async {
-        var streamKey = _buildStreamKey(
-          endpoint: methodStreamCallContext.fullEndpointPath,
-          method: methodStreamCallContext.method.name,
-          methodStreamId: methodStreamId,
+          // The stream complete message is sent as an error to circumvent
+          // branching when passing along stream events to the the handler.
+          () => outputController.addError(_StreamComplete()),
```

Comment:
At least on the face of it this appears to play more nice with the underlying stream! :+1: